src/classifyInputString.cpp

```diff
@@ -0,0 +1,81 @@
+#include "classifyInputString.h"
+
+namespace {
+
+    InputParam classifyInputStringBase(const std::string & str) {
+
+        if(str.empty())
+            return unknown_param;
+
+        // if exactly 64 chars in length, it is likely a transaction id
+        if(str.length() == 64)
+            return txid_param;
+
+        // if it starts with certain chars, and is of a certain length, it may be a bitcoin address
+        if(str[0] == '1' || str[0] == '3' || str[0] == 'm' || str[0] == 'n' || str[0] == '2')
+            if(str.length() >= 26 && str.length() < 36)
+                return address_param;
+
+        // before testing for various txrefs, get rid of any unknown characters
+        std::string s = bech32::stripUnknownChars(str);
+
+        using namespace txref::limits;
```

Comment:
I think this txref part is the only code we'd want to propose in the BIP.  Our usage of txids and addresses will not be relevant to the txref spec.

Reply:
BTW, I'm noticing that applying the comment to the first relevant line results in a confusing github preview.  I'll try and adapt to their default preview of four lines, or learn more about selecting lines that I'm commenting on.

Reply:
Good point. I can split out the address/txid code into another function so we can narrow the important part to just the txref handling.